pkg/products/amqonline/prometheusRules.go

```diff
@@ -0,0 +1,70 @@
+package amqonline
+
+import (
+	"context"
+	"fmt"
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/config"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *Reconciler) reconcileKubeStateMetricsAmqOnline(ctx context.Context,serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error){
+
+	monitoringConfig := config.NewMonitoring(config.ProductConfig{})
+ 	rule := &monitoringv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:		"ksm-amqonline-alerts",
+			Namespace:  r.Config.GetNamespace(),
+		},
+}
+
+	var namespace = r.Config.GetNamespace()
+
+	rules := []monitoringv1.Rule{
+		{
+			Alert: "AMQOnlinePodCount",
+			Annotations: map[string]string{
+				"sop_url": "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md",
+				"message": "Pod count for namespace {{  $labels.namespace  }} is {{  $value }}. Expected at least 2 pods.",
+			},
+			Expr:   intstr.FromString(fmt.Sprintf("(1-absent(kube_pod_status_ready{condition='true', namespace='%s'})) or sum(kube_pod_status_ready{condition='true', namespace='%s'}) < 2", namespace, namespace)),
+			For:    "5m",
+			Labels: map[string]string{"severity": "critical"},
+		},
+		{
+			Alert: "AMQOnlineContainerHighMemory",
+			Annotations: map[string]string{
+				"sop_url": "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md",
+				"message": "The {{  $labels.container  }} Container in the {{ $labels.pod  }} Pod has been using {{  $value }}% of available memory for longer than 15 minutes.",
+			},
+			Expr:   intstr.FromString(fmt.Sprintf("sum by(container, pod) (container_memory_usage_bytes{container!='',namespace='%s'}) / sum by(container, pod) (kube_pod_container_resource_limits_memory_bytes{namespace='%s'}) * 100 > 90", namespace, namespace)),
+			For:    "15m",
+			Labels: map[string]string{"severity": "warning"},
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, serverClient, rule, func() error {
```

Comment:
Would recommend getting the operation result and logging it in the events of ```controllerutil.OperationResultNone```. Would be easier to debug if it had created or updated, during a reconcile.

Reply:
I think the error message that is returned should be enough to figure out why an error is occurring I dont really see the need to know if it happened during and update or create

Reply:
The information of either a create or update, being logged is crucial because: when we upgrade an existing cluster and expect a dashboard to be updated (rather than created ) during a reconcile, it would be nice to know that it was updated. In a similar way for creation. It is not about the error , rather than a log for us to track if it was created or updated as expected. And code changes like this would be ideal, so that it doesnt flood the messages during each reconcile: ```	if opRes != controllerutil.OperationResultNone {
		r.Logger.Infof("operation result of XXXXXX  %v was %v",, opRes)
	}```. It would log only created or updated cases, which ideally should occur only once.

Reply:
Scratch the above comment looking through how other create or updates functions are done in the code this morning implement the log.  I can no see why its needed so im happy to add it in